### PR TITLE
Plugin openstack

### PR DESCRIPTION
**Pull request by genaumann**
This PR contains the openstack driver for molecule.

Local tests were successfull.

### CODE REVIEW

File: src/molecule_plugins/openstack/playbooks/create.yml
Comment (konstruktoid) [inline src/molecule_plugins/openstack/playbooks/create.yml:104]: adding volume support which is needed if no-disk flavors are used would be good, imo

```
diff --git a/src/molecule_plugins/openstack/playbooks/create.yml b/src/molecule_plugins/openstack/playbooks/create.yml
index 7d01a7d..f71f5c8 100644
--- a/src/molecule_plugins/openstack/playbooks/create.yml
+++ b/src/molecule_plugins/openstack/playbooks/create.yml
@@ -102,6 +102,10 @@
         image: "{{ item.image }}"
         key_name: "{{ key_name }}"
         flavor: "{{ item.flavor }}"
+        volume_size: "{{ item.volume_size | default(omit) }}"
+        boot_from_volume: "{{ item.boot_from_volume | default(false) }}"
+        terminate_volume: "{{ item.terminate_volume | default(false) }}"
+        auto_ip: "{{ item.auto_ip | default(true) }}"
         network: >-
           {{
             'molecule-test-' + item.network.name + '-' + uuid
```

Comment (genaumann) [reply src/molecule_plugins/openstack/playbooks/create.yml:104]: Yes, you are right. I will add this in the next few days

**Comment by genaumann** [comment]
@konstruktoid volume support added.

Could you help to fix the "tox / lint" job?

**Comment by konstruktoid** [comment]
The lint job will be fixed with https://github.com/ansible-community/molecule-plugins/pull/159, bit I don't have permissions to add or remove tags.

**Comment by konstruktoid** [comment]
> volume support added.
 
Awesome, will test as soon as possible

File: src/molecule_plugins/openstack/playbooks/create.yml
Comment (konstruktoid) [inline src/molecule_plugins/openstack/playbooks/create.yml:114]: Why?

**Comment by genaumann** [comment]
@konstruktoid I refactored the volume creation.

**Comment by konstruktoid** [comment]
https://github.com/ansible-community/molecule/issues/3919 will probably impact this

**Comment by genaumann** [comment]
@konstruktoid what is the current status? Do you need anything else so that the PR can be merged?

**Comment by konstruktoid** [comment]
Just that any conflicts and lints need to be fixed, and then another reviewer with merge permissions needs to give a thumbs up.

**Comment by juanlufont** [comment]
> > The lint job will be fixed with #159, but I don't have permissions to add or remove tags.
> 
> I thought the lint job is fixed with #159. Local linting was successfull. I will have a look at the merge conflict.

Hi!

I locally merged your OpenStack plugin code with the molecule-plugins upstream "main" branch to check the code against the latest lint configuration, I got only 2 errors:

```
yaml[octal-values]: Forbidden implicit octal value "0600"
test/roles/openstackplugin/molecule/default/create.yml:36

yaml[octal-values]: Forbidden implicit octal value "0600"
test/roles/openstackplugin/molecule/default/destroy.yml:23
```

After using quotes around those `0600`  permission strings the code passed all the latest lint tests. I can create a PR against your fork, but the changes are so minimal that you may apply them directly.

File: doc/openstack/README.rst
Comment (zhan9san) [inline doc/openstack/README.rst:28]: This command is not available now.

**Review by zhan9san — CHANGES_REQUESTED** [review]

File: doc/openstack/README.rst
Comment (juanlufont) [reply doc/openstack/README.rst:28]: I am not the author of the PR but I would like to help to get the OpenStack plugin merged in this repository. I would not mind to patch this PR and create a new one (not sure if there is a better way to do this).

Comment (zhan9san) [reply doc/openstack/README.rst:28]: @juanlufont 

Thanks for your attention.

A new PR is fine for me. I'll review it.

**Pull request closed without merging**
